gitpod-io/gitpod
# Add phone verification

## Description
This PR adds phone number verification before untrusted users can start a workspace.

A user is considered trusted if
 - Their GitHub account is older than 30 days, or
 - Their GitLab account `confirmed_at` event is older than 30 days, or
 - Their Gitpod account has been created before 2022-08-24 (i.e. existing users)

## Related Issue(s)
Fixes #11339 

See also https://github.com/gitpod-io/ops/pull/4149

## How to test
*Test 1*
 - Create a fresh GitHub account and sign in (use admin to unblock). 
 - You should need to do phone number verification before you can start your first workspace.
 - Once the number is verified you should be able to work as usual.

*Test 2*
 - Use a GitHub account older than 30 days.
 - You should be able to start a workspace without verification.

*Test 1 ++*
 - delete the `lastVerificationTimestamp` in your user's additional Data.
 - if the account had started a workspace before, you should not need to verify
 - if the account doesn't have any workspaces, you should be asked to verify

## Release Notes

```release-note
Added phone number verification in SaaS, to help mitigate abuse.
```

## Documentation


## Werft options:
